File: KargoPaylasimJava/app/src/main/java/com/example/kargopaylasimjava/view/RegisterActivity.java
package com.example.kargopaylasimjava.view;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.activity.ComponentActivity;
import androidx.lifecycle.ViewModelProvider;

import com.example.kargopaylasimjava.R;
import com.example.kargopaylasimjava.di.AppContainer;
import com.example.kargopaylasimjava.dto.AuthDtos;
import com.example.kargopaylasimjava.factory.VMFactories;
import com.example.kargopaylasimjava.model.UiState;
import com.example.kargopaylasimjava.viewmodel.AuthViewModel;

public class RegisterActivity extends ComponentActivity {

    private AuthViewModel vm;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_register);

        AppContainer container = new AppContainer(getApplicationContext());
        vm = new ViewModelProvider(this, new VMFactories.AuthVmFactory(container.repo, container.tokenStore))
                .get(AuthViewModel.class);

        EditText etFirst = findViewById(R.id.etFirst);
        EditText etLast  = findViewById(R.id.etLast);
        EditText etPhone = findViewById(R.id.etPhone);
        EditText etTc    = findViewById(R.id.etTc);
        EditText etPass  = findViewById(R.id.etPassword);

        EditText etAddrTitle = findViewById(R.id.etAddressTitle);
        EditText etCity      = findViewById(R.id.etCity);
        EditText etDistrict  = findViewById(R.id.etDistrict);
        EditText etNeighborhood = findViewById(R.id.etNeighborhood);
        EditText etAddrLine  = findViewById(R.id.etAddressLine);
        EditText etPostal    = findViewById(R.id.etPostal);

        Button btn = findViewById(R.id.btnRegister);
        ProgressBar prog = findViewById(R.id.progress);

        btn.setOnClickListener(v -> {
            String neighborhood = etNeighborhood.getText().toString().trim();
            if (neighborhood.isEmpty()) neighborhood = null;

            String postal = etPostal.getText().toString().trim();
            if (postal.isEmpty()) postal = null;

            AuthDtos.RegisterReq req = new AuthDtos.RegisterReq(
                    etFirst.getText().toString().trim(),
                    etLast.getText().toString().trim(),
                    etPhone.getText().toString().trim(),
                    etTc.getText().toString().trim(),
                    etPass.getText().toString(),
                    etAddrTitle.getText().toString().trim(),
                    etCity.getText().toString().trim(),
                    etDistrict.getText().toString().trim(),
                    neighborhood,
                    etAddrLine.getText().toString().trim(),
                    postal
            );

            // Kotlin'de register(req) var; istersen auto-login için registerAndSetup(req) kullan
            vm.register(req);
        });

        vm.registerState.observe(this, st -> {
            if (st instanceof UiState.Loading) {
                prog.setVisibility(View.VISIBLE);
            } else if (st instanceof UiState.Success) {
                prog.setVisibility(View.GONE);
                Toast.makeText(this, "Kayıt başarılı", Toast.LENGTH_LONG).show();
                startActivity(new Intent(this, LoginActivity.class));
                finish();
            } else if (st instanceof UiState.Error) {
                prog.setVisibility(View.GONE);
                Toast.makeText(this, ((UiState.Error<?>) st).message, Toast.LENGTH_LONG).show();
            } else {
                prog.setVisibility(View.GONE);
            }
        });
    }
}

File: KargoPaylasimKotlin/app/src/main/java/com/example/kargopaylasimkotlin/view/RegisterActivity.kt
package com.example.kargopaylasimkotlin.view

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import com.example.kargopaylasimkotlin.R
import android.view.View
import android.widget.*
import androidx.activity.ComponentActivity
import androidx.lifecycle.ViewModelProvider
import com.example.kargopaylasimkotlin.di.AppContainer
import com.example.kargopaylasimkotlin.dto.AddressCreateReq
import com.example.kargopaylasimkotlin.dto.RegisterReq
import com.example.kargopaylasimkotlin.factory.AuthVmFactory
import com.example.kargopaylasimkotlin.model.UiState
import com.example.kargopaylasimkotlin.viewmodel.AuthViewModel

class RegisterActivity : ComponentActivity() {

    private lateinit var vm: AuthViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_register)

        val container = AppContainer(applicationContext)
        vm = ViewModelProvider(this, AuthVmFactory(container.repo, container.tokenStore))
            .get(AuthViewModel::class.java)

        val etFirst = findViewById<EditText>(R.id.etFirst)
        val etLast = findViewById<EditText>(R.id.etLast)
        val etPhone = findViewById<EditText>(R.id.etPhone)
        val etTc = findViewById<EditText>(R.id.etTc)
        val etPass = findViewById<EditText>(R.id.etPassword)
        val btn = findViewById<Button>(R.id.btnRegister)
        val prog = findViewById<ProgressBar>(R.id.progress)
        val tvBack = findViewById<TextView>(R.id.tvBackLogin)

        tvBack.setOnClickListener { finish() }

        btn.setOnClickListener {
            val first = etFirst.text.toString().trim()
            val last = etLast.text.toString().trim()
            val phone = etPhone.text.toString().trim()
            val tc = etTc.text.toString().trim()
            val pass = etPass.text.toString()

            val titleAddr = findViewById<EditText>(R.id.etAddressTitle).text.toString().trim()
            val city = findViewById<EditText>(R.id.etCity).text.toString().trim()
            val district = findViewById<EditText>(R.id.etDistrict).text.toString().trim()
            val neigh = findViewById<EditText>(R.id.etNeighborhood).text.toString().trim()
            val line = findViewById<EditText>(R.id.etAddressLine).text.toString().trim()
            val postal = findViewById<EditText>(R.id.etPostal).text.toString().trim()

            if (first.isBlank() || last.isBlank() || phone.isBlank() || tc.length != 11 || pass.length < 4) {
                Toast.makeText(this, "Bilgileri kontrol et (TC 11 hane, şifre min 4).", Toast.LENGTH_LONG).show()
                return@setOnClickListener
            }

            // neighborhood opsiyonelse burada zorunlu yapma
            if (titleAddr.isBlank() || city.isBlank() || district.isBlank() || line.isBlank()) {
                Toast.makeText(this, "Adres bilgileri eksik.", Toast.LENGTH_LONG).show()
                return@setOnClickListener
            }

            val regReq = RegisterReq(
                first_name = first,
                last_name = last,
                phone = phone,
                tc_no = tc,
                password = pass,
                address_title = titleAddr,
                city = city,
                district = district,
                neighborhood = neigh.takeIf { it.isNotBlank() },
                address_line = line,
                postal_code = postal.takeIf { it.isNotBlank() }
            )

            val addrReq = AddressCreateReq(
                title = titleAddr,
                city = city,
                district = district,
                neighborhood = neigh.takeIf { it.isNotBlank() },
                address_line = line,
                postal_code = postal.takeIf { it.isNotBlank() }
            )

            vm.registerAndSetup(regReq, addrReq)
        }


        vm.registerState.observe(this) { st ->
            when (st) {
                is UiState.Loading -> prog.visibility = View.VISIBLE
                is UiState.Success -> {
                    prog.visibility = View.GONE
                    Toast.makeText(this, "Kayıt başarılı. Giriş yapabilirsiniz.", Toast.LENGTH_LONG).show()
                    finish()
                }
                is UiState.Error -> {
                    prog.visibility = View.GONE
                    Toast.makeText(this, st.message, Toast.LENGTH_LONG).show()
                }
                else -> prog.visibility = View.GONE
            }
        }
    }
}
